tests/csm/rest/test_iam_users.py

```diff
@@ -2127,6 +2126,190 @@ def test_39217(self):
         assert_utils.assert_true(len(diff_items) == 0, "Capabilities are not updated properly")
         self.log.info("##### Test completed -  %s #####", test_case_name)
 
+    @pytest.mark.csmrest
+    @pytest.mark.lc
+    @pytest.mark.cluster_user_ops
+    @pytest.mark.parallel
+    @pytest.mark.tags('TEST-38971')
+    def test_38971(self):
+        """
+        Verify create IAM user with uid same as tenant
+        """
+        test_case_name = cortxlogging.get_frame()
+        self.log.info("##### Test started -  %s #####", test_case_name)
+        self.log.info("Step-1: Creating IAM user")
+        payload = self.csm_obj.iam_user_payload_rgw("valid")
+        resp = self.csm_obj.create_iam_user_rgw(payload)
+        self.log.info("Verify Response : %s", resp)
+        assert resp.status_code == HTTPStatus.CREATED, "IAM user creation failed"
+        uid = resp.json()["tenant"] + "$" + resp.json()["user_id"]
+        self.created_iam_users.add(uid)
+
+        self.log.info("Step-2: Creating IAM user with same name as tenant")
+        payload = self.csm_obj.iam_user_payload_rgw("valid")
+        payload.update({"tenant": uid})
```

Comment:
as uid is modified above now, will that work for issue which you saw OR you need user_id only here without resp.json()["tenant"] + "$"

Reply:
Thanks for the catch. I was giving problem. Fixed now.